examples/images_contours_and_fields/multi_image.py

```diff
@@ -4,80 +4,46 @@
 ===========
 
 Make a set of images with a single colormap, norm, and colorbar.
-
-It also illustrates colorbar tick labelling with a multiplier.
 """
 
-from matplotlib.pyplot import figure, show, sca, sci
-from matplotlib import cm, colors
-from matplotlib.font_manager import FontProperties
+from matplotlib import colors, pyplot as plt
 import numpy as np
 
+np.random.seed(19680801)
 Nr = 3
 Nc = 2
+cmap = "cool"
 
-fig = figure()
-cmap = cm.cool
-
-figtitle = 'Multiple images'
-t = fig.text(0.5, 0.95, figtitle,
-             horizontalalignment='center',
-             fontproperties=FontProperties(size=16))
-
-cax = fig.add_axes([0.2, 0.08, 0.6, 0.04])
+fig, axs = plt.subplots(Nr, Nc)
+fig.suptitle('Multiple images')
 
-w = 0.4
-h = 0.22
-ax = []
 images = []
-vmin = 1e40
-vmax = -1e40
 for i in range(Nr):
     for j in range(Nc):
-        pos = [0.075 + j * 1.1 * w, 0.18 + i * 1.2 * h, w, h]
-        a = fig.add_axes(pos)
-        if i > 0:
-            a.set_xticklabels([])
-        # Make some fake data with a range that varies
-        # somewhat from one plot to the next.
+        # Generate data with a range that varies from one plot to the next.
         data = ((1 + i + j) / 10) * np.random.rand(10, 20) * 1e-6
-        dd = data.ravel()
-        # Manually find the min and max of all colors for
-        # use in setting the color scale.
-        vmin = min(vmin, np.min(dd))
-        vmax = max(vmax, np.max(dd))
-        images.append(a.imshow(data, cmap=cmap))
-
-        ax.append(a)
-
-# Set the first image as the master, with all the others
-# observing it for changes in cmap or norm.
-
-
-class ImageFollower(object):
-    'update image in response to changes in clim or cmap on another image'
-
-    def __init__(self, follower):
-        self.follower = follower
-
-    def __call__(self, leader):
-        self.follower.set_cmap(leader.get_cmap())
-        self.follower.set_clim(leader.get_clim())
-
+        images.append(axs[i, j].imshow(data, cmap=cmap))
+        axs[i, j].label_outer()
 
+# Find the min and max of all colors for use in setting the color scale.
+vmin = min(image.get_array().min() for image in images)
+vmax = max(image.get_array().max() for image in images)
 norm = colors.Normalize(vmin=vmin, vmax=vmax)
-for i, im in enumerate(images):
+for im in images:
     im.set_norm(norm)
-    if i > 0:
-        images[0].callbacksSM.connect('changed', ImageFollower(im))
 
-# The colorbar is also based on this master image.
-fig.colorbar(images[0], cax, orientation='horizontal')
+fig.colorbar(images[0], ax=axs, orientation='horizontal', fraction=.1)
 
-# We need the following only if we want to run this interactively and
-# modify the colormap:
+# Make images respond to changes in the norm of other images, but be careful
+# not to recurse infinitely!
+def update(changed_image):
```

Comment:
pep8 is whining about two blank line above and below this function

Reply:
should be placated now...